src/test/java/co/cask/hydrator/salesforce/etl/SalesforceBatchSourceETLTest.java

```diff
@@ -0,0 +1,118 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.salesforce.etl;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+
+
+/**
+ * {@inheritDoc}
+ */
+public class SalesforceBatchSourceETLTest extends BaseSalesforceBatchSourceETLTest {
+  @Test
+  public void testTypesConversion() throws Exception {
+    String query = "SELECT Id, IsDeleted, Type, Probability, ExpectedRevenue, TotalOpportunityQuantity, " +
+      "LastActivityDate, LastModifiedDate FROM Opportunity";
+    List<StructuredRecord> records = getResultsBySOQLQuery(query);
+    Assert.assertNotEquals(0, records.size());
+
+    StructuredRecord record = records.get(0);
+
+    Schema expectedSchema = Schema.recordOf("output",
+                                            Schema.Field.of("Id",
+                                                            Schema.of(Schema.Type.STRING)),
+                                            Schema.Field.of("IsDeleted",
+                                                            Schema.of(Schema.Type.BOOLEAN)),
+                                            Schema.Field.of("Type",
+                                                            Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                                            Schema.Field.of("Probability",
+                                                            Schema.nullableOf(Schema.of(Schema.Type.DOUBLE))),
+                                            Schema.Field.of("ExpectedRevenue",
+                                                            Schema.nullableOf(Schema.of(Schema.Type.DOUBLE))),
+                                            Schema.Field.of("TotalOpportunityQuantity",
+                                                            Schema.nullableOf(Schema.of(Schema.Type.DOUBLE))),
+                                            Schema.Field.of("LastActivityDate",
+                                                            Schema.nullableOf(Schema.of(Schema.LogicalType.DATE))),
+                                            Schema.Field.of("LastModifiedDate",
+                                                            Schema.of(Schema.LogicalType.TIMESTAMP_MILLIS))
+    );
+
+    Assert.assertEquals(expectedSchema, record.getSchema());
+  }
+
+  @Test
+  public void testValuesReturned() throws Exception {
+    String query = "SELECT OwnerId, Name, IsDeleted FROM Account"; // select only non-nillable fields
```

Comment:
You should insert data before test then fetch data by id or by expected field values to be sure that inserted data fetched properly. Please add datasets for each case and test all supported Salesforce data types.